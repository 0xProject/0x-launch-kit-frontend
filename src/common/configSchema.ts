import { Schema } from 'jsonschema';

export const configSchema: Schema = {
    id: '/configSchema',
    properties: {
        tokens: { type: 'array', items: { $ref: '/tokenMetaDataSchema' } },
        pairs: { type: 'array', items: { $ref: '/pairSchema' } },
    },
    required: ['tokens', 'pairs'],
    type: 'object',
};

export const schemas: Schema[] = [
    {
        id: '/addressSchema',
        type: 'string',
        pattern: '^0x[0-9a-f]{40}$',
    },
    {
        id: '/wholeNumberSchema',
        anyOf: [
            {
                type: 'string',
                pattern: '^\\d+$',
            },
            {
                type: 'integer',
            },
        ],
    },
    {
        id: '/marketFilterSchema',
        properties: {
            text: { type: 'string' },
            valute: { type: 'string' },
        },
        required: ['text', 'value'],
        type: 'object',
    },
    {
        id: '/pairSchema',
        properties: {
            base: { type: 'string' },
            quote: { type: 'string' },
        },
        required: ['base', 'quote'],
        type: 'object',
    },
    {
        id: '/tokenMetaDataSchema',
        properties: {
            symbol: { type: 'string' },
            name: { type: 'string' },
            icon: { type: 'string' },
            primaryColor: { type: 'string' },
            decimals: { $ref: '/wholeNumberSchema' },
            displayDecimals: { $ref: '/wholeNumberSchema' },
        },
        required: ['decimals', 'symbol', 'name', 'addresses'],
        type: 'object',
    },
    {
        id: '/configSchema',
        properties: {
            tokens: { type: 'array', items: { $ref: '/tokenMetaDataSchema' } },
            pairs: { type: 'array', items: { $ref: '/pairSchema' } },
            marketFilters: { type: 'array', items: { $ref: '/marketFilterSchema' } },
        },
        required: ['tokens', 'pairs'],
        type: 'object',
    },
];
