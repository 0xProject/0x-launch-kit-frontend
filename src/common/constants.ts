import { BigNumber } from '0x.js';

import { Network } from '../util/types';

export const ERC20_APP_BASE_PATH = '/erc20';
export const ERC721_APP_BASE_PATH = '/erc721';
export const DEFAULT_BASE_PATH = process.env.REACT_APP_DEFAULT_BASE_PATH || ERC20_APP_BASE_PATH;

export const RELAYER_URL = process.env.REACT_APP_RELAYER_URL || 'http://localhost:3001/api/v2';

export const TX_DEFAULTS = {
    gasLimit: 1000000,
};

export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';

export const FEE_RECIPIENT = process.env.REACT_APP_FEE_RECIPIENT || ZERO_ADDRESS;

export const ETH_DECIMALS = 18;

export const UI_DECIMALS_DISPLAYED_ON_STEP_MODALS = 3;
export const UI_DECIMALS_DISPLAYED_SPREAD_PERCENT = 2;
export const UI_DECIMALS_DISPLAYED_DEFAULT_PRECISION = 2;
export const UI_DECIMALS_DISPLAYED_ORDER_SIZE = 0;
export const UI_DECIMALS_DISPLAYED_PRICE_ETH = 7;
export const UI_GENERAL_TITLE = 'Launch Kit';

export const METAMASK_EXTENSION_URL = 'https://metamask.io/';
export const METAMASK_CHROME_EXTENSION_DOWNLOAD_URL =
    'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn';

// Default value is enabled, 0 is disabled
export const UI_UPDATE_CHECK_INTERVAL: number = process.env.REACT_APP_UI_UPDATE_CHECK_INTERVAL
    ? Number.parseInt(process.env.REACT_APP_UI_UPDATE_CHECK_INTERVAL as string, 10)
    : 5000;

// Default value is enabled, 0 is disabled
export const UPDATE_ETHER_PRICE_INTERVAL: number = process.env.REACT_APP_UPDATE_ETHER_PRICE_INTERVAL
    ? Number.parseInt(process.env.REACT_APP_UPDATE_ETHER_PRICE_INTERVAL as string, 10)
    : 3600000;

export const NOTIFICATIONS_LIMIT: number =
    Number.parseInt(process.env.REACT_APP_NOTIFICATIONS_LIMIT as string, 10) || 20;

export const GWEI_IN_WEI = new BigNumber(1000000000);

export const ONE_MINUTE_MS = 1000 * 60;

export const DEFAULT_GAS_PRICE = GWEI_IN_WEI.multipliedBy(6);

export const DEFAULT_ESTIMATED_TRANSACTION_TIME_MS = ONE_MINUTE_MS * 2;

export const GIT_COMMIT: string = process.env.REACT_APP_GIT_COMMIT || '';

export const START_BLOCK_LIMIT: number = Number.parseInt(process.env.REACT_APP_START_BLOCK_LIMIT as string, 10) || 1000;

export const LOGGER_ID: string = process.env.REACT_APP_LOGGER_ID || 'veridex';

export const ERC20_THEME_NAME: string = process.env.REACT_APP_ERC20_THEME_NAME || 'DARK_THEME';

export const ERC721_THEME_NAME: string = process.env.REACT_APP_ERC721_THEME_NAME || 'LIGHT_THEME';

export const COLLECTIBLES_SOURCE: string = process.env.REACT_APP_COLLECTIBLES_SOURCE || 'mocked';

export const COLLECTIBLE_NAME: string = process.env.REACT_APP_COLLECTIBLE_NAME || 'Unknown';
export const COLLECTIBLE_DESCRIPTION: string = process.env.REACT_APP_COLLECTIBLE_DESCRIPTION || 'Unknown';

const mockERC721Address = '0x07f96aa816c1f244cbc6ef114bb2b023ba54a2eb'; // Mock ERC721 in ganache
export const COLLECTIBLE_ADDRESS = process.env.REACT_APP_COLLECTIBLE_ADDRESS || mockERC721Address;

export const STEP_MODAL_DONE_STATUS_VISIBILITY_TIME: number =
    Number.parseInt(process.env.REACT_APP_STEP_MODAL_DONE_STATUS_VISIBILITY_TIME as string, 10) || 3500;

export const OPENSEA_API_KEY = process.env.REACT_APP_OPENSEA_API_KEY;

export const NETWORK_ID: number = Number.parseInt(process.env.REACT_APP_NETWORK_ID as string, 10) || Network.Mainnet;

export const NETWORK_NAME: string = Network[NETWORK_ID];

export const FILLS_LIMIT: number =
    Number.parseInt(process.env.REACT_APP_FILLS_LIMIT as string, 10) || 50;

