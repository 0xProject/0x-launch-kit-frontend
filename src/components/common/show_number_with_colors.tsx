import { BigNumber } from '0x.js';
import React from 'react';
import styled from 'styled-components';

import { padRightSplitted } from '../../util/number_utils';

interface ShowNumberWithColorsProps {
    num: BigNumber;
    isHover?: boolean;
}

interface SpanRightProps {
    isHover?: boolean;
}

class ShowNumberWithColors extends React.Component<ShowNumberWithColorsProps, {}> {
    public render = () => {
        const { num, isHover } = this.props;
        const numSplitted = padRightSplitted(num);
        const SpanLeft = styled.span`
            color: ${props => props.theme.componentsTheme.textColorCommon};
        `;
        const SpanRight = styled.span<SpanRightProps>`
            color: ${props =>
                props.isHover
                    ? props.theme.componentsTheme.textColorCommon
                    : props.theme.componentsTheme.numberDecimalsColor};
        `;

        return (
            <>
                <SpanLeft>{numSplitted.num}</SpanLeft>
                <SpanRight isHover={isHover}>{numSplitted.diff}</SpanRight>
            </>
        );
    };
}

export { ShowNumberWithColors };
