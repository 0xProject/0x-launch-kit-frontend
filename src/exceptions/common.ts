export const INSUFFICIENT_MAKER_BALANCE_ERR = 'INSUFFICIENT MAKER BALANCE';
export const SIGNATURE_ERR = 'User denied message signature';
export const USER_DENIED_TRANSACTION_SIGNATURE_ERR = 'User denied transaction signature';
export const INSUFFICIENT_ORDERS_TO_FILL_AMOUNT_ERR = 'There are not enough orders to fill this amount';
export const RELAYER_ERR = 'There was an error with the relayer';
export const INSUFFICIENT_FEE_BALANCE = 'INSUFFICIENT MAKER FEE BALANCE';
export const INSUFFICIENT_FEE_BALANCE_MSG = `You don't have enough 0x to pay fees`;
export const INSUFFICIENT_ETH_BALANCE_FOR_DEPOSIT = 'INSUFFICIENT_ETH_BALANCE_FOR_DEPOSIT';
export const UNEXPECTED_ERROR = 'An unexpected error happened';
