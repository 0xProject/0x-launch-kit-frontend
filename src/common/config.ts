import { Validator } from 'jsonschema';

import configFile from '../config.json';
import { ConfigFile } from '../util/types';

import { configSchema, schemas } from './configSchema';

export class Config {
    private static _instance: Config;
    private readonly _validator: Validator;
    private readonly _config: ConfigFile;
    public static getInstance(): Config {
        if (!Config._instance) {
            Config._instance = new Config();
        }
        return Config._instance;
    }
    public static getConfig(): ConfigFile {
        return this.getInstance()._config;
    }
    constructor() {
        this._validator = new Validator();
        for (const schema of schemas) {
            this._validator.addSchema(schema, schema.id);
        }
        this._validator.validate(configFile, configSchema, { throwError: true });
        this._config = configFile;
    }
}
