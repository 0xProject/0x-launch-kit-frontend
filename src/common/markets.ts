import { Filter } from '../util/types';

import { Config } from './config';

export const availableMarkets = Config.getConfig().pairs;
const allFilter = {
    text: 'ALL',
    value: null,
};
const suppliedMarketFilters = Config.getConfig().marketFilters;
export const marketFilters: Filter[] = suppliedMarketFilters ? [...suppliedMarketFilters, allFilter ] : [];
