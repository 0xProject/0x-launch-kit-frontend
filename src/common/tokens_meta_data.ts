import { Config } from './config';

export interface TokenMetaData {
    addresses: { [key: number]: string };
    symbol: string;
    decimals: number;
    name: string;
    primaryColor: string;
    id?: string;
    icon?: string;
    displayDecimals?: number;
}
export const KNOWN_TOKENS_META_DATA: TokenMetaData[] = Config.getConfig().tokens;
