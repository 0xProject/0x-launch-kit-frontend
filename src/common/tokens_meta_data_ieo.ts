import { ConfigIEO } from './config';

export interface TokenIEOMetaData {
    addresses: { [key: number]: string };
    symbol: string;
    decimals: number;
    name: string;
    primaryColor: string;
    id?: string;
    c_id?: string;
    icon?: string;
    displayDecimals?: number;
    minAmount?: number;
    maxAmount?: number;
    precision?: number;
    description?: string;
    website?: string;
    verisafe_sticker?: string;
    owners?: string[];
    social?: {
        facebook_url?: string;
        reddit_url?: string;
        twitter_url?: string;
        telegram_url?: string;
        discord_url?: string;
        bitcointalk_url?: string;
        youtube_url?: string;
        medium_url?: string;
    };
    feePercentage?: string;
    endDate?: number | string;
}

export const KNOWN_TOKENS_IEO_META_DATA: TokenIEOMetaData[] = ConfigIEO.getConfig().tokens;
