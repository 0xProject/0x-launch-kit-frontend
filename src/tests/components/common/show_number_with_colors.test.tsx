import { BigNumber } from '0x.js';
import { shallow } from 'enzyme';
import React from 'react';

import { ShowNumberWithColors } from '../../../components/common/show_number_with_colors';

describe('ShowNumberWithColors', () => {
    it('should be initialized with value 123.0000', () => {
        // given
        const value = new BigNumber('123');

        // when
        const wrapper = shallow(<ShowNumberWithColors num={value} />);

        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should be initialized with value 123.2200', () => {
        // given
        const value = new BigNumber('123.22');

        // when
        const wrapper = shallow(<ShowNumberWithColors num={value} />);

        // then
        expect(wrapper).toMatchSnapshot();
    });
});
