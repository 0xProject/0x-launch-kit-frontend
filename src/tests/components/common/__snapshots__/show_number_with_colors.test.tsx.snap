// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowNumberWithColors should be initialized with value 123.0000 1`] = `
<Fragment>
  <styled.span>
    123
  </styled.span>
  <styled.span>
    .0000
  </styled.span>
</Fragment>
`;

exports[`ShowNumberWithColors should be initialized with value 123.2200 1`] = `
<Fragment>
  <styled.span>
    123.22
  </styled.span>
  <styled.span>
    00
  </styled.span>
</Fragment>
`;
